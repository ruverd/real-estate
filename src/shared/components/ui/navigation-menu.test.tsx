import { render, screen } from "@testing-library/react";
import {
  NavigationMenu,
  NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuTrigger
} from "./navigation-menu";

describe("NavigationMenu", () => {
  it("should renders navigation menu with children", () => {
    render(
      <NavigationMenu>
        <NavigationMenuList>
          <NavigationMenuItem>
            <NavigationMenuTrigger>Item 1</NavigationMenuTrigger>
            <NavigationMenuContent>Content 1</NavigationMenuContent>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu> 
    );

    expect(screen.getByText("Item 1")).toBeInTheDocument();
  });

  it("should renders navigation menu without viewport when viewport prop is false", () => {
    const { container } = render(
      <NavigationMenu viewport={false}>
        <NavigationMenuList>
          <NavigationMenuItem>
            <NavigationMenuTrigger>Item 1</NavigationMenuTrigger>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu>
    );

    expect(container.querySelector('[data-slot="navigation-menu-viewport"]')).not.toBeInTheDocument();
  });

  it("should renders navigation menu link with correct styling", () => {
    render(
      <NavigationMenu>
        <NavigationMenuLink className="test-class">
          Link Text
        </NavigationMenuLink>
      </NavigationMenu>
    );

    const link = screen.getByText("Link Text");

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("data-slot", "navigation-menu-link");
    expect(link).toHaveClass("test-class");
  });

  it("should renders navigation menu trigger with chevron icon", () => {
    render(
      <NavigationMenu>
        <NavigationMenuList>
          <NavigationMenuItem>
            <NavigationMenuTrigger>Trigger Text</NavigationMenuTrigger>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu>
    );

    expect(screen.getByText("Trigger Text")).toBeInTheDocument();
    
    const svg = document.querySelector("svg");

    expect(svg).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <NavigationMenu>
        <NavigationMenuList>
          <NavigationMenuItem>
            <NavigationMenuTrigger>Item 1</NavigationMenuTrigger>
            <NavigationMenuContent>Content 1</NavigationMenuContent>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu> 
    );
    
    expect(container).toMatchSnapshot();
  });
});
