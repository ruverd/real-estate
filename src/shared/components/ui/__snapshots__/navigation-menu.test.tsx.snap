// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu > should match snapshot 1`] = `
<div>
  <nav
    aria-label="Main"
    class="group/navigation-menu relative flex max-w-max flex-1 items-center justify-center"
    data-orientation="horizontal"
    data-slot="navigation-menu"
    data-viewport="true"
    dir="ltr"
  >
    <div
      style="position: relative;"
    >
      <ul
        class="group flex flex-1 list-none items-center justify-center gap-1"
        data-orientation="horizontal"
        data-slot="navigation-menu-list"
        dir="ltr"
      >
        <li
          class="relative"
          data-slot="navigation-menu-item"
        >
          <button
            aria-controls="radix-:r7:-content-radix-:r8:"
            aria-expanded="false"
            class="group inline-flex h-9 w-max items-center justify-center rounded-md bg-background px-4 py-2 text-sm font-medium hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground disabled:pointer-events-none disabled:opacity-50 data-[state=open]:hover:bg-accent data-[state=open]:text-accent-foreground data-[state=open]:focus:bg-accent data-[state=open]:bg-accent/50 ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1 group"
            data-radix-collection-item=""
            data-slot="navigation-menu-trigger"
            data-state="closed"
            id="radix-:r7:-trigger-radix-:r8:"
          >
            Item 1
             
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down relative top-[1px] ml-1 size-3 transition duration-300 group-data-[state=open]:rotate-180"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="absolute top-full left-0 isolate z-50 flex justify-center"
    />
  </nav>
</div>
`;
